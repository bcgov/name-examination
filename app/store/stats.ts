import { getStats } from '~/util/namex-api'
import { getFormattedDateWithTime } from '~/util/date'
import { sortNameChoices } from '~/util'
import { type NameRequest } from '~/types'
import { defineStore } from 'pinia'

export const useStatsStore = defineStore('stats', () => {
  /**
   * Fetches paged statistics data based on the provided timespan and myStats flag.
   *
   * @param {string | null} timespan - The timespan for the statistics data. If null, the default timespan is used.
   * @param {boolean} myStats - A flag indicating whether to fetch statistics related to user's own stats.
   * @return {Promise<any>} A promise that resolves with the fetched statistics data.
   * @throws {Error} Throws an error if the fetching process fails.
   */
  async function getPagedStats (timespan: string | null, myStats: boolean): Promise<any> {
    const params = new URLSearchParams({
      timespan: timespan || '1',
      currentpage: '1',
      perpage: '1000'
    })
    if (myStats) params.append('myStats', 'True')

    try {
      const response = await getStats(params)

      if (response.ok) {
        const jsonData = await response.json() // Parse response JSON

        if (jsonData.numRecords >= 0) {
          const requestsData = jsonData.nameRequests as Array<NameRequest>

          const processedData = requestsData.map((statsData:any) => {
            statsData.names = sortNameChoices(statsData.names)
            if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
              statsData.comments[0].timestamp = getFormattedDateWithTime(statsData.comments[0].timestamp)
            }
            statsData.lastUpdate = getFormattedDateWithTime(statsData.lastUpdate)
            statsData.names.forEach((name:any) => {
              if (name.decision_text) {
                name.decision_text = name.decision_text.split(/\n\n/)
              }
            })
            return statsData
          })
          return {
            nameRequests: processedData,
            numRecords: jsonData.numRecords
          }
        }
      } else {
        throw new Error('Network response was not ok')
      }
    } catch (error) {
      console.error('There was a problem with your fetch operation:', error)
      throw error // Rethrow the error to handle it elsewhere if needed
    }
  }

  return {
    getPagedStats
  }
})
