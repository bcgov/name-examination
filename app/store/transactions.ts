import {
  ConsentFlag,
  type EntityTypeCode,
  type RequestActionCode,
  type RequestTypeCode,
} from '~/enums/codes'
import { Status } from '~/enums/nr-status'
import {
  type NameChoice,
  type NameRequest,
  type TransactionEntry,
  type Transactions,
} from '~/types'
import { emitter } from '~/util/emitter'
import { getNameRequest, getTransactions } from '~/util/namex-api'
import { toMappedRequestType } from '~/util/request-type'
import requestTypes from '~/data/request_types.json'
import { sortNameChoices } from '~/util'

export const useTransactions = defineStore('transactions', () => {
  const nr = ref<NameRequest>()
  const transactions = ref<Array<TransactionEntry>>([])
  const loadingNr = ref(false)
  const loadingTransactions = ref(false)

  async function loadNr(nrNumber: string) {
    loadingNr.value = true
    const nrResponse = await getNameRequest(nrNumber)
    nr.value = await nrResponse.json()
    loadingNr.value = false
  }

  async function loadTransactions(nrNumber: string) {
    loadingTransactions.value = true
    const transactionsResponse = await getTransactions(nrNumber)
    const transactionsJson = (await transactionsResponse.json()) as Transactions
    transactionsJson.transactions.forEach(t => sortNameChoices(t.names))
    transactions.value = transactionsJson.transactions
    loadingTransactions.value = false
  }

  async function initialize(nrNumber: string) {
    try {
      await loadNr(nrNumber)
      await loadTransactions(nrNumber)
    } catch (e) {
      emitter.emit('error', {
        title: 'Transactions Error',
        message: 'Failed to retrieve transactions data.',
      })
    } finally {
      loadingNr.value = false
      loadingTransactions.value = false
    }
  }

  function getStatusDisplay(status: Status | null, names: Array<NameChoice>) {
    if (!status) return 'N/A'

    let displayState = status as string
    if (status === Status.Conditional) {
      displayState = 'CONDITIONAL APPROVED'
    } else if (status === Status.Consumed && names && names.length > 0) {
      const approvedName = names.find((name) =>
        [Status.Approved, Status.Condition].includes(name.state)
      )
      displayState =
        approvedName?.state === Status.Condition
          ? 'CONDITIONAL APPROVED'
          : 'APPROVED'
      displayState += ` / Used for ${approvedName?.corpNum}`
    }
    return displayState
  }

  function getRequestTypeDisplay(
    requestType: RequestTypeCode,
    action: RequestActionCode
  ) {
    if (!nr.value) return 'N/A'
    const mapped = toMappedRequestType(
      requestType,
      action,
      nr.value.entity_type_cd
    )
    return requestTypes.find((r) => r.value == mapped)?.text || 'N/A'
  }

  function getConsentDisplay(date: string | null, consentFlag: ConsentFlag) {
    return date
      ? 'Required. Received.'
      : consentFlag === ConsentFlag.Required
      ? 'Required. Not yet received.'
      : 'Not required'
  }

  return {
    nr,
    transactions,
    loadingNr,
    loadingTransactions,
    initialize,
    getStatusDisplay,
    getRequestTypeDisplay,
    getConsentDisplay,
  }
})
