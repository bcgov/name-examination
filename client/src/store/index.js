/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
