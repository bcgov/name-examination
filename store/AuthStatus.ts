import { defineStore } from 'pinia';

export const useAuthStore = defineStore('auth', {
  state: () => ({
    auth: false,
  }),
  actions: {
    toggleAuth() {
      this.auth = !this.auth;
    },
  },
});
