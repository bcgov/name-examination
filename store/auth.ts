import ConfigHelper from '../util/config-helper'
import KeycloakServices from '../public/keycloak/keycloak'
import { SessionStorageKeys } from '../util/constants'
import { defineStore } from 'pinia'

interface AuthState {
  token: string;
  idToken: string;
  refreshToken: string;
  kcGuid: string;
  loginSource: string;
}

export const useAuthStore = defineStore({
  id: 'auth',
  state: (): AuthState => ({
    token: '',
    idToken: '',
    refreshToken: '',
    kcGuid: '',
    loginSource: ''
  }),
  getters: {
    isAuthenticated (state): boolean {
      return !!state.token
    },
    keycloakGuid (state): string {
      return state.kcGuid || KeycloakServices.getUserInfo().keycloakGuid
    },
    currentLoginSource (state): string {
      return state.loginSource || KeycloakServices.getUserInfo().loginSource
    }
  },
  actions: {
    setKCToken (token: string): void {
      this.token = token
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakToken, token)
    },
    setIDToken (idToken: string): void {
      this.idToken = idToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakIdToken, idToken)
    },
    setRefreshToken (refreshToken: string): void {
      this.refreshToken = refreshToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakRefreshToken, refreshToken)
    },
    setKCGuid (kcGuid: string): void {
      this.kcGuid = kcGuid
    },
    setLoginSource (loginSource: string): void {
      this.loginSource = loginSource
    },
    clearSession (): void {
      // Reset all state properties related to the user's session
      this.token = ''
      this.idToken = ''
      this.refreshToken = ''
      this.kcGuid = ''
      this.loginSource = ''

      // Clear the session storage values
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakIdToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakRefreshToken)
    },
    syncWithSessionStorage (): void {
      const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken) || ''
      const idToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakIdToken) || ''
      const refreshToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakRefreshToken) || ''

      this.token = token
      this.idToken = idToken
      this.refreshToken = refreshToken
    }
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useAuthStore, import.meta.hot))
}
