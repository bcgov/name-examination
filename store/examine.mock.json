{
  "trademarkJSON": {
    "names": [
      {
        "application_number": "300000102764600",
        "description": "(1) Fruit drinks and soft drinks containing fruit juices",
        "name": "STEEP",
        "score": 82.06702,
        "status": "Registration published"
      },
      {
        "application_number": "300000167106400",
        "description": "(1) Structural thermal energy efficient panels for residential, commercial and industrial construction as exterior/interior walls, floors, ceilings and roofs. (2) Energy efficient windows.",
        "name": "STEEP",
        "score": 82.06702,
        "status": "Registration published"
      },
      {
        "application_number": "300000055451300",
        "description": "(1) Games, namely board games, parlour games, puzzles, strategy games, card games.",
        "name": "CO-OPERATIVE",
        "score": 78.5403,
        "status": "Registration published"
      }
    ]
  },
  "historiesJSON": {
    "highlighting": "",
    "names": [
      {
        "jurisdiction": "BC",
        "name": "NOT SO STEEP COOPERATIVE",
        "name_state_type_cd": "R",
        "nr_num": "NR 5979354",
        "score": 11.4911,
        "start_date": "2008-09-16T16:43:17Z",
        "submit_count": 1
      },
      {
        "jurisdiction": "BC",
        "name": "NOT SO STEEP COOPERATIVE",
        "name_state_type_cd": "A",
        "nr_num": "NR 3252689",
        "score": 11.4911,
        "start_date": "2008-09-16T16:41:29Z",
        "submit_count": 1
      }
    ],
    "response": {
      "maxScore": 11.4911,
      "name": "not so steep cooperative",
      "numFound": 2,
      "rows": "50",
      "start": 0
    }
  },
  "parseConditions": [
    {
      "allow_use": "Y",
      "consent_required": "N",
      "id": 4421,
      "instructions": " ",
      "text": "Use Of This Term Is Restricted Under The Cooperative Association Act R.S.B.C. 1999 C. 28, and only given for a Cooperative. ",
      "consent_required_tf": false,
      "allow_use_tf": true,
      "phrase": "COOPERATIVE"
    }
  ],
  "parsedSynonymConflicts": [
    {
      "text": "SO",
      "highlightedText": "<span class=\"stem-highlight\">SO</span>",
      "meta": "PROXIMITY SEARCH",
      "children": [
        {
          "text": "XYZ SO LTD.",
          "highlightedText": "XYZ <span class=\"stem-highlight\">SO</span> LTD.",
          "jurisdiction": "BC",
          "nrNumber": "0685772",
          "startDate": "2004-01-22T00:00:00Z",
          "source": "CORP"
        },
        {
          "text": "SO COOL INC.",
          "highlightedText": "<span class=\"stem-highlight\">SO</span> COOL INC.",
          "jurisdiction": "BC",
          "nrNumber": "NR 0769877",
          "startDate": "2006-09-25T00:00:00Z",
          "source": "NR"
        },
        {
          "text": "JOHNNY SO INC.",
          "highlightedText": "JOHNNY <span class=\"stem-highlight\">SO</span> INC.",
          "jurisdiction": "ON",
          "nrNumber": "A4312694",
          "startDate": "2006-09-25T00:00:00Z",
          "source": "CORP"
        }
      ]
    },
    {
      "text": "SO*",
      "highlightedText": "<span class=\"stem-highlight\">SO</span>*",
      "meta": "EXACT WORD ORDER",
      "children": []
    }
  ],
  "macros": [
    {
      "id": 4,
      "name": "Amend ",
      "reason": "*** Name Amended To Allow Acceptance ***"
    },
    {
      "id": 3,
      "name": "Amend Real Estate ",
      "reason": "***Name Amended To Allow Acceptance By Real Estate Council***"
    },
    {
      "id": 5,
      "name": "Assumed Name",
      "reason": "Assumed Name A Foreign Entity Whose Name Has Been Rejected For Use In Bc, May Register In Bc Under An Assumed Name."
    },
    {
      "id": 6,
      "name": "Assumed Yes ",
      "reason": "A Foreign Entity That Is Registering In British Columbia As An Extraprovincial Company And Adopting An Assumed Name Must Provide The Registrar With A Covering Letter Attaching An Undertaking To Carry On Business Under The Assumed Name.  Sample Working For The Undertaking Can Be Found On Page 34 Of The Information For Registration Of An Extraprovincial Company In British Columbia   Information Package And Also In The Online Help Text At Www.Corporateonline.Gob Please Fax The Letter Containing The Undertaking To The Attention Of The Corporations Unit, Bc Registry Services (Fax Number:250 356-8923)."
    }
  ],
  "comments": [
    {
      "comment": "This NR was RESET.",
      "examiner": "someone@idir",
      "id": 136253,
      "timestamp": "2023-05-29T22:02:56.288783+00:00"
    },
    {
      "comment": "another comment",
      "examiner": "someone@idir",
      "id": 138311,
      "timestamp": "2024-01-15T20:02:59.297305+00:00"
    },
    {
      "comment": "new comment",
      "examiner": "someone@idir",
      "id": 138310,
      "timestamp": "2024-01-15T20:02:47.354556+00:00"
    },
    {
      "comment": "This NR was RESET.",
      "examiner": "someone@idir",
      "id": 137890,
      "timestamp": "2023-11-06T22:26:27.902815+00:00"
    }
  ],

  "conflicts": [
    {
      "type": "corp",
      "startDate": "2004-01-22",
      "nrNumber": "0685772",
      "jurisdiction": "BC",
      "text": "XYZ SO LTD.",
      "invalidRecordInd": false,
      "incorp #": "0685772",
      "directors": ["ADA SO", "Paul James"],
      "nature of business": "Not available",
      "records office delivery address": [
        "6407 CYPRESS STREET",
        "VANCOUVER BC CA V6M 3S4"
      ],
      "registered office delivery address": [
        "6407 CYPRESS STREET",
        "VANCOUVER BC CA V6M 3S4"
      ]
    },
    {
      "type": "name",
      "startDate": "2006-09-25",
      "nrNumber": "NR 0769877",
      "jurisdiction": "BC",
      "text": "SO COOL INC.",
      "invalidRecordInd": false,
      "applicants": {
        "firstName": "John",
        "lastName": "Test",
        "addrLine1": "6407 CYPRESS STREET",
        "addrLine2": "",
        "addrLine3": "",
        "city": "Vancouver",
        "stateProvinceCd": "BC",
        "postalCd": "V6M 3S4",
        "countryTypeCd": "CA",
        "phoneNumber": "305-343-3434",
        "emailAddress": "test@example.com",
        "clientFirstName": "John",
        "clientLastName": "Test",
        "contact": "John Test"
      },
      "state": "APPROVED",
      "comments": [
        {
          "comment": "here is a comment",
          "examiner": "anonymous@idir",
          "timestamp": "2024-01-15T20:02:59.297305+00:00"
        },
        {
          "comment": "here is another comment",
          "examiner": "anonymous@idir",
          "timestamp": "2023-11-06T22:26:27.902815+00:00"
        }
      ],
      "names": [
        {
          "name": "SO COOL INC.",
          "decision_text": "accepted",
          "state": "APPROVED",
          "conflict1": "ADA SO LTD."
        }
      ]
    },
    {
      "type": "corp",
      "startDate": "2006-09-25",
      "nrNumber": "A4312694",
      "jurisdiction": "ON",
      "text": "JOHNNY SO INC.",
      "invalidRecordInd": false,
      "incorp #": "A4312694",
      "attorney names": "Not Available",
      "nature of business": "Not available",
      "directors": ["ADA SO", "Paul James"],
      "head office": ["6407 CYPRESS STREET", "TORONTO ON CA L4S 3S4"]
    }
  ],
  "compName1": {
    "choice": 1,
    "name": "STEEPHILL COOP",
    "state": "REJECTED",
    "decision_text": "Rejected due to conflict with...",
    "conflict1": null,
    "conflict2": null,
    "conflict3": null,
    "consumptionDate": null,
    "corpNum": null,
    "conflict1_num": null,
    "conflict2_num": null,
    "conflict3_num": null,
    "comment": null
  },
  "compName2": {
    "choice": 2,
    "name": "NOT SO STEEP COOPERATIVE",
    "state": "NE",
    "decision_text": null,
    "conflict1": null,
    "conflict2": null,
    "conflict3": null,
    "consumptionDate": null,
    "corpNum": null,
    "conflict1_num": null,
    "conflict2_num": null,
    "conflict3_num": null,
    "comment": null
  },
  "compName3": {
    "choice": 3,
    "name": "BLACK HILL PERSONAL COOP COOPERATIVE",
    "state": "NE",
    "decision_text": null,
    "conflict1": null,
    "conflict2": null,
    "conflict3": null,
    "consumptionDate": null,
    "corpNum": null,
    "conflict1_num": null,
    "conflict2_num": null,
    "conflict3_num": null,
    "comment": null
  },
  "additionalInfo": "Here's some additional info. This one is made long so we can see what it looks like when additional info is long. So here's some more text. And some more.",
  "natureOfBusiness": "Here's our nature of business. This one is made long so we can see what it looks like when nature of business is long. So here's some more text. And some more.",
  "clientFirstName": "Jane",
  "clientLastName": "Frost",
  "firstName": "Jack",
  "middleName": "",
  "lastName": "Frost",
  "addressLine1": "3434 Main Street",
  "addressLine2": "Suite 303",
  "addressLine3": "Top Floor",
  "city": "Saywarn",
  "province": "BC",
  "postalCode": "v8v8v8",
  "country": "CA",
  "phone": "305-343-3434",
  "fax": "306-343-3333",
  "conEmail": "test@gov.bc.ca",
  "contactName": "Jack Frost"
}
