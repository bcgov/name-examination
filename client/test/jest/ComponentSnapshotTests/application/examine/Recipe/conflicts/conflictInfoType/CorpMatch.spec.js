/* eslint-disable */
import { shallowMount } from '@vue/test-utils';
import CorpMatch from '@/components/application/Examine/Recipe/conflicts/conflictInfoType/corpMatch';
import store from '@/store'

describe('CorpMatch.vue', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(CorpMatch, {store: store});
    });

   it("renders a CorpMatch component", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
