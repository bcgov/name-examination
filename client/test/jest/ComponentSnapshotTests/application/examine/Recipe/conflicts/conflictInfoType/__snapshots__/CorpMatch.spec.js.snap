// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorpMatch.vue renders a CorpMatch component 1`] = `
<span>
  <h3
    class="corpType"
  >
    BC Corporation
  </h3>
   
  <h3>
    Incorporation Number
  </h3>
   
  <p>
    
  </p>
   
  <h3>
    Incorporation Date
  </h3>
   
  <p>
    
  </p>
   
  <h3>
    Directors
  </h3>
    
  <h3>
    Jurisdiction
  </h3>
   
  <p>
    
  </p>
   
  <h3>
    Nature Of Business
  </h3>
   
  <p>
    
  </p>
   
  <h3>
    Records Office Delivery Address
  </h3>
    
  <h3>
    Registered Office Delivery Address
  </h3>
   
</span>
`;
